### PR TITLE
Add Log Shipping rule

### PR DESCRIPTION
**Pull request by stevelle**
Add example definition needed by rsyslog client to ship logs to
logstash. Example defines logstash_port as needed by the log shipping
rule.

Add play to run all logging related tasks and related README change.

### CODE REVIEW

File: etc/openstack_deploy/user_extras_variables.yml [hunk 1]
```diff
@@ -78,3 +78,22 @@ maas_filesystem_threshold: 90.0
...
 # elasticsearch_heap_size_mb: 1024
+
+# Ship all logs to the logstash container as well as the rsyslog_server.
+# Uncomment the following if the setup-logging play is going to be used.
+# logstash_port: 5544
+# rsyslog_client_user_defined_targets:
+#    - name: "rax-logging"
+#      proto: "tcp"
+#      port: "{{ logstash_port }}"
+#      hostname: "groups['rsyslog_all'][0]"
```
Comment (odyssey4me) [inline etc/openstack_deploy/user_extras_variables.yml]: something appears to be wrong here, this is processed too literally and I'm not entirely sure about the intent here

File: etc/openstack_deploy/user_extras_variables.yml [hunk 2]
```diff
@@ -78,3 +78,22 @@ maas_filesystem_threshold: 90.0
 # For an AIO it's recommended to set the following to limit the expected RAM
 # usage of elasticsearch.
 # elasticsearch_heap_size_mb: 1024
+
+# Ship all logs to the logstash container as well as the rsyslog_server.
+# Uncomment the following if the setup-logging play is going to be used.
+# logstash_port: 5544
+# rsyslog_client_user_defined_targets:
+#    - name: "rax-logging"
```
Comment (odyssey4me) [inline etc/openstack_deploy/user_extras_variables.yml]: rpc-logging is perhaps better?

File: playbooks/site.yml [hunk 3]
```diff
@@ -1,5 +1,4 @@
----
```
Comment (odyssey4me) [inline playbooks/site.yml]: I don't think this should be removed, and would think that the copyright gumpf should be here too.

File: etc/openstack_deploy/user_extras_variables.yml (hunk 1 above)
Comment (stevelle) [inline etc/openstack_deploy/user_extras_variables.yml]: the hostname was a bit of a WIP.  iterating on that bit now.

File: etc/openstack_deploy/user_extras_variables.yml (hunk 2 above)
Comment (stevelle) [inline etc/openstack_deploy/user_extras_variables.yml]: done

File: playbooks/site.yml (hunk 3 above)
Comment (stevelle) [inline playbooks/site.yml]: done

File: etc/openstack_deploy/user_extras_variables.yml [hunk 4]
```diff
@@ -78,3 +78,22 @@ maas_filesystem_threshold: 90.0
...
 # elasticsearch_heap_size_mb: 1024
+
+# Ship all logs to the logstash container as well as the rsyslog_server.
+# Uncomment the following if the setup-logging play is going to be used.
+# logstash_port: 5544
+# rsyslog_client_user_defined_targets:
+#    - name: "rpc-logging"
+#      proto: "tcp"
+#      port: "{{ logstash_port }}"
+#      hostname: "groups['logstash_all'][0]"
```
Comment (d34dh0r53) [inline etc/openstack_deploy/user_extras_variables.yml]: This needs to have double curly braces:
    hostname: "{{ groups['logstash_all'][0] }}"

**Comment by d34dh0r53** [comment]
With the exception of the double curly problem I think this is good to go.

File: etc/openstack_deploy/user_extras_variables.yml (hunk 4 above)
Comment (stevelle) [inline etc/openstack_deploy/user_extras_variables.yml]: done

**Pull request closed without merging**